packages/squiggle-lang/src/value/index.ts

```diff
@@ -416,3 +514,60 @@ export type Value =
   | VScale
   | VDomain
   | VVoid;
+
+export const valueTypeName = (v: Value) => {
+  const names = {
+    Array: "Array",
+    Bool: "Boolean",
+    Date: "Date",
+    Dist: "Distribution",
+    Lambda: "Function",
+    Number: "Number",
+    String: "String",
+    Dict: "Dictionary",
+    TimeDuration: "Time Duration",
+    Plot: "Plot",
+    TableChart: "Table Chart",
+    Scale: "Scale",
+    Domain: "Domain",
+    Void: "Void",
+  } as const;
+  return names[v.type] || "Unknown";
```

Comment:
So, we have:
- `type` on `VFoo` classes
- `tag` on `SqFoo` classes (which is guaranteed to be the same as `type` by `SqAbstractValue` base class)
- `heading` on `VariableBox` on components, which is usually `tag` from `SqFoo`, but not always
- and now this list of names

Can we reuse this? somehow Should we expose `valueTypeName` through `SqValue`?

If we go with my suggestion about `publicName()`, then we could proxy it through `SqAbstractValue` base class, and then call it in `ExpressionViewer` in this line, instead of `value.tag`:

```typescript
  const heading = boxProps.heading || value.tag;
```